Update Skeleton Instructions for Development Guide Project Setup

dv4harr10 opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR will update the Documentation Guide Project Setup markdown files instructions to create new skeleton repositories or skeletonize existing repositories.  

## 💭 Motivation and context ##

These updates are required for uniformity of the cisagov skeleton updates to create and update repositories and to advise on the latest skeletonize tools used.

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Create a release.

dv4harr10 committed:
Update skeleton instructions for README

This commit will remove out-dated
skeleton tool instructions. Previously
the documentation was using a legacy
skeleton tool. Now the instructions
will reflect using the gh CLI
skeleton tool extension. Links
have been updated along with steps
to make a repository from a
skeleton.

dv4harr10 committed:
Add description change for gh skeleton tool

Updated naming of gh tool to be more
accurate verbiage from gh CLI extension
to gh skeleton. Updated link to skeleton AWS lambda python which was linked to archived skeleton AWS lambda.

jsf9k commented on project_setup/README.md:
```diff
@@ -3,10 +3,9 @@
-a new local repository. Once you've cloned and configured the repository
-to your local machine, it can be published to a repository created on GitHub.
+The [`gh-skeleton`](https://github.com/cisagov/gh-skeleton)
+tool provides the ability to easily start new projects from the
```
```suggestion
GitHub CLI (`gh`) extension provides the ability to easily start new projects from the
```

jsf9k commented on project_setup/README.md:
```diff
@@ -38,14 +37,13 @@ branch protection - [see our branch protection guide for details](branch-protect
 [skeletonize an existing repository](skeletonize-existing-repository.md).
 
-## Using the skeleton tool to start a new repository 💀🛠 ##
+## Using the gh-skeleton tool to start a new repository 💀🛠 ##
```
```suggestion
## Using the gh-skeleton `gh` extension to start a new repository 💀🛠 ##
```

jsf9k commented on project_setup/README.md:
```diff
@@ -19,7 +18,7 @@ branch protection - [see our branch protection guide for details](branch-protect
 
 - [Adding a skeleton configuration to an existing repository](#adding-a-skeleton-configuration-to-an-existing-repository)
-- [Using the skeleton tool to start a new repository 💀🛠](#using-the-skeleton-tool-to-start-a-new-repository-)
+- [Using the gh-skeleton tool to start a new repository 💀🛠](#using-the-gh-skeleton-tool-to-start-a-new-repository-)
```
```suggestion
- [Using the gh-skeleton `gh` extension to start a new repository 💀🛠](#using-the-gh-skeleton-gh-extension-to-start-a-new-repository-)
```

jsf9k commented on project_setup/README.md:
```diff
@@ -38,14 +37,13 @@ branch protection - [see our branch protection guide for details](branch-protect
-a new local repository.  Once you've cloned and configured the repository
-to your local machine, you then publish it to a repository created on GitHub.
+in this organization.  The [`gh-skeleton`](https://github.com/cisagov/gh-skeleton)
+tool provides the ability to easily start new projects from the existing
```
```suggestion
`gh` extension provides the ability to easily start new projects from the existing
```

jsf9k commented on project_setup/README.md:
```diff
@@ -87,19 +85,20 @@ skeleton-aws-lambda
-your new repository for publication.  The tool will print out each command it
-is issuing and its result.
+Next, use the [`gh-skeleton`](https://github.com/cisagov/gh-skeleton)
+tool to clone, rename, and prepare the contents of your new repository
```
```suggestion
`gh` extension to clone, rename, and prepare the contents of your new repository
```

jsf9k commented on project_setup/README.md:
```diff
@@ -145,7 +144,7 @@ Use the following commands to push the new repository to github:
 ## Create and publish the GitHub repository ##
 
-Once the `skeleton` tool has run, the next step is to publish to a GitHub
+Once the `gh-skeleton` tool has run, the next step is to publish to a GitHub
```
```suggestion
Once the `gh-skeleton` `gh` extension has run, the next step is to publish to a GitHub
```

jsf9k commented on project_setup/skeleton-list.md:
```diff
@@ -4,23 +4,24 @@ We maintain a number of [`cisagov` skeleton projects](https://github.com/search?
-directory of the [development-guide](https://github.com/cisagov/development-guide)
-repository.
+The [`gh-skeleton`](https://github.com/cisagov/gh-skeleton) is
+a custom tool that provides the ability to easily start new projects
```
```suggestion
a GitHub CLI (`gh`) extension that provides the ability to easily start new projects
```

jsf9k commented on project_setup/skeleton-list.md:
```diff
@@ -4,23 +4,24 @@ We maintain a number of [`cisagov` skeleton projects](https://github.com/search?
 [skeletonize an existing repository](skeletonize-existing-repository.md).
 - To skeletonize a new repository, please see the guide to
-[start a new repository from a skeleton](new-repository-from-skeleton.md).
+[start a new repository from a skeleton](https://github.com/cisagov/gh-skeleton).
```
This is a better link:
```suggestion
[start a new repository from a skeleton](README.md).
```

jsf9k commented on project_setup/skeletonize-existing-repository.md:
```diff
@@ -1,12 +1,14 @@
-to keep the repository updated and standardized.
+and enables our [Lineage GitHub Action](https://github.com/cisagov/action-lineage/)
+to keep the repository updated and standardized. The purpose of the
+instruction here is to skeletonize an existing non-skeletonized
```
```suggestion
instructions here is to describe how to skeletonize an existing non-skeletonized
```

jsf9k commented on project_setup/skeletonize-existing-repository.md:
```diff
@@ -15,23 +17,24 @@ appropriate for the major languages that we use. This lets us standardize
-1. [Add the skeleton as a remote](#add-a-skeleton-as-remote) to the
-non-skeletonized repository
+1. [Add the skeleton as a remote](#add-the-skeleton-as-a-remote) to the
+non-skeletonized existing repository
```
```suggestion
existing non-skeletonized repository
```

jsf9k commented on project_setup/skeletonize-existing-repository.md:
```diff
@@ -98,21 +101,21 @@ from merging in destructive upstream changes.
 
 The skeleton will bring along with it our standard pre-commit hook
-configurations, including linting and other checks, with `setup-env`.
+configurations, including linting and other checks, within `setup-env`.
```
The previous language was correct:
```suggestion
configurations, including linting and other checks, with `setup-env`.
```

dv4harr10 committed:
Apply suggestions from code review

Update naming convention to GitHub CLI 'gh' Extension

Co-authored-by: Shane Frasier <jeremy.frasier@gwe.cisa.dhs.gov>

dv4harr10 committed:
Link Correction

Co-authored-by: Shane Frasier <jeremy.frasier@gwe.cisa.dhs.gov>

dv4harr10 committed:
Language verbiage corrections

Co-authored-by: Shane Frasier <jeremy.frasier@gwe.cisa.dhs.gov>

jsf9k committed:
Reformat to placate Markdown linter

dav3r reviewed (approved):
Approved, but with one small item noted.

dav3r commented on project_setup/skeletonize-existing-repository.md:
````diff
@@ -63,10 +66,10 @@ up the prerequisites, this involves:
 pre-commit install
 ```
 
-## Fix merge conflicts ##
+## Fix all the inevitable merge conflicts ##
````
Somehow, this comment didn't get recorded when I put in my review.  I think this change should be reverted to the original text and the link change above that references this heading should also be reverted.

jsf9k replied:
Fixed in commit ae5f047.

dv4harr10 replied:
Hi Dave @dav3r, did we resolve the conversation? Also, I'm not sure if I have permissions to merge the PR?

dav3r replied:
Yes, thank you for making the change in https://github.com/cisagov/development-guide/commit/ae5f047ebfc0ea7687ffecd1ee132cecc0d951df.  If you don't have permissions to merge this PR, one of the members of our dev team can take care of it.

dv4harr10 committed:
Fix text and link for merge conflict

jsf9k commented:
Let's get @mcdonnnj's review before merging.

mcdonnnj commented on project_setup/README.md:
```diff
@@ -3,10 +3,9 @@
-helper tool included in the [`scripts`](scripts) directory can quickly setup
-a new local repository. Once you've cloned and configured the repository
-to your local machine, it can be published to a repository created on GitHub.
+The [`gh-skeleton`](https://github.com/cisagov/gh-skeleton) GitHub CLI
```
It looks like we link out to the `gh-skeleton` repository in multiple places. Please switch this to a reference-style link and update all uses of "gh-skeleton" to both use this link when appropriate and be formatted consistently.

dv4harr10 replied:
Hi @mcdonnnj , update to reference-style link. Addressed at 077cfbf

mcdonnnj commented on project_setup/README.md:
```diff
@@ -19,7 +18,7 @@ branch protection - [see our branch protection guide for details](branch-protect
 
 - [Adding a skeleton configuration to an existing repository](#adding-a-skeleton-configuration-to-an-existing-repository)
-- [Using the skeleton tool to start a new repository 💀🛠](#using-the-skeleton-tool-to-start-a-new-repository-)
+- [Using the gh-skeleton `gh` extension to start a new repository 💀🛠](#using-the-gh-skeleton-gh-extension-to-start-a-new-repository-)
```
The emoji no longer make sense.

```suggestion
- [Using the gh-skeleton `gh` extension to start a new repository](#using-the-gh-skeleton-gh-extension-to-start-a-new-repository-)
```

dv4harr10 replied:
Hi @mcdonnnj , removed skeleton tool emoji. Addressed at ffbea4a
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

mcdonnnj commented on project_setup/README.md:
````diff
@@ -54,7 +52,7 @@ for your new repository. For a list of available skeletons, see the
+gh skeleton list
 ```
 
 ```console
````
Please combine these codeblocks and ensure the sample output aligns with the output of the extension. Please also update the lead-in formatting like so:

```
```console
$ gh skeleton list
```

dv4harr10 replied:
Hi @mcdonnnj , updated, replace text at line 54 to 58 Readme.md. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

mcdonnnj commented on project_setup/README.md:
````diff
@@ -87,19 +85,20 @@ skeleton-aws-lambda
 
 ```console
-./skeleton clone [options] <parent-repo-name> <new-repo-name>
+gh skeleton clone [options] <parent-repo-name> <new-repo-name>
````
```suggestion
$ gh skeleton clone [options] <parent-repo-name> <new-repo-name>
```

dv4harr10 replied:
Hi @mcdonnnj , I received 2 linter failed errors for adding the dollar sign prompt, it complains that expected output is required. I worked with Michael on this solution to add the following: 
```console
$ gh skeleton clone [options] <parent-repo-name> <new-repo-name>
# Expected output:
# Cloning into 'new-repo-name' .  
```
Alternatively we may need to omit the dollar sign prompt. What do you suggest?

dv4harr10 replied:
Hi @mcdonnnj , the change above satisfied the linter. Addressed at 077cfbf

mcdonnnj commented on project_setup/README.md:
````diff
@@ -87,19 +85,20 @@ skeleton-aws-lambda
 For example, to create a project based on `skeleton-ansible-role` named
 `ansible-role-quantum-rng` in your local `~/projects` directory:
 
 ```bash
````
```suggestion
```console
```

dv4harr10 replied:
Hi @mcdonnnj , updated text line 100. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

mcdonnnj commented on project_setup/README.md:
````diff
@@ -87,19 +85,20 @@ skeleton-aws-lambda
 
 ```bash
-./skeleton clone --change-dir ~/projects skeleton-ansible-role ansible-role-quantum-rng
+gh skeleton clone --change-dir ~/projects skeleton-ansible-role ansible-role-quantum-rng
````
```suggestion
$ gh skeleton clone --change-dir ~/projects skeleton-ansible-role ansible-role-quantum-rng
```

mcdonnnj commented on project_setup/skeleton-list.md:
```diff
@@ -4,23 +4,24 @@ We maintain a number of [`cisagov` skeleton projects](https://github.com/search?
-The [`skeleton`](scripts/skeleton) helper tool is included in the [`scripts`](scripts)
-directory of the [development-guide](https://github.com/cisagov/development-guide)
-repository.
+The [`gh-skeleton`](https://github.com/cisagov/gh-skeleton) is
```
This doesn't make sense as it is written. Also please use a reference-style link and convert all uses of "gh-skeleton" in line with my previous suggestion.

```suggestion
The [`gh-skeleton`](https://github.com/cisagov/gh-skeleton) project is
```

mcdonnnj commented on project_setup/skeleton-list.md:
```diff
@@ -4,23 +4,24 @@ We maintain a number of [`cisagov` skeleton projects](https://github.com/search?
-repository.
+The [`gh-skeleton`](https://github.com/cisagov/gh-skeleton) is
+a GitHub CLI (`gh`) extension that provides the ability to easily start new projects
+from existing library of skeleton repositories.
```
```suggestion
from an existing library of skeleton repositories.
```

mcdonnnj commented on project_setup/skeleton-list.md:
```diff
@@ -4,23 +4,24 @@ We maintain a number of [`cisagov` skeleton projects](https://github.com/search?
 [skeletonize an existing repository](skeletonize-existing-repository.md).
 - To skeletonize a new repository, please see the guide to
-[start a new repository from a skeleton](new-repository-from-skeleton.md).
+[start a new repository from a skeleton](README.md).
```
Are we able to use a link fragment here to link to the appropriate section?

mcdonnnj commented on project_setup/skeleton-list.md:
```diff
@@ -4,23 +4,24 @@ We maintain a number of [`cisagov` skeleton projects](https://github.com/search?
-After you've run `setup-env` in this repository, the `skeleton` tool will be
-available to you via the command line. To see a list of available skeletons,
-use the following command:
+After you've installed the [`gh` CLI](https://github.com/cli/cli) followed
```
If we're going to link out I would prefer we use a reference-style link and update all uses of "gh" to use it where appropriate.

```suggestion
After you've installed [`gh`](https://github.com/cli/cli) followed
```

mcdonnnj commented on project_setup/skeleton-list.md:
````diff
@@ -4,23 +4,24 @@ We maintain a number of [`cisagov` skeleton projects](https://github.com/search?
+the tool will be available to you via the command line. To see a list of
+available skeletons, use the following command:
 
 ```bash
````
```suggestion
```console
```

mcdonnnj commented on project_setup/skeleton-list.md:
````diff
@@ -4,23 +4,24 @@ We maintain a number of [`cisagov` skeleton projects](https://github.com/search?
 
 ```bash
-skeleton list
+gh skeleton list
````
```suggestion
$ gh skeleton list
```

mcdonnnj commented on project_setup/skeletonize-existing-repository.md:
```diff
@@ -1,12 +1,14 @@
 # Skeletonize an Existing Repository #
 
 Skeletonizing a repository standardizes our development setup and environment,
```
```suggestion
Skeletonizing a repository standardizes our development setup and environment
```

dv4harr10 replied:
Hi @mcdonnnj , removed comma at end. of line. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

mcdonnnj commented on project_setup/skeletonize-existing-repository.md:
```diff
@@ -1,12 +1,14 @@
-to keep the repository updated and standardized.
+and enables our [Lineage GitHub Action](https://github.com/cisagov/action-lineage/)
+to keep the repository updated and standardized. The purpose of the
+instructions here is to describe how to skeletonize an existing non-skeletonized
```
```suggestion
instructions here are to describe how to skeletonize an existing non-skeletonized
```

dv4harr10 replied:
Hi @mcdonnnj , grammar text update. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

mcdonnnj commented on project_setup/skeletonize-existing-repository.md:
```diff
@@ -15,23 +17,24 @@ appropriate for the major languages that we use. This lets us standardize
-non-skeletonized repository
+1. [Add the skeleton as a remote](#add-the-skeleton-as-a-remote) to the
+existing non-skeletonized repository
 1. Pull with `--allow-unrelated-histories`
```
```suggestion
1. Pull with `git pull --allow-unrelated-histories skeleton-parent`
```

dv4harr10 replied:
Hi @mcdonnnj , suggested working update for command. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

mcdonnnj commented on project_setup/skeletonize-existing-repository.md:
```diff
@@ -15,23 +17,24 @@ appropriate for the major languages that we use. This lets us standardize
+First, decide which of the available skeletons best fits your
+existing non-skeletonized repository. To see a list of available
+skeletons, use the `gh skeleton list` command or see
 the [list of skeletons](skeleton-list.md).
```
Are we able to use a link fragment here to link to the appropriate section?

dv4harr10 replied:
Hi @mcdonnnj , Can you please elaborate regarding the fragment link required ? From my understanding the 'gh skeleton list' command is to be run at the command line. And the skeleton-list link goes to skeleton-list.md document.

dv4harr10 replied:
Hi @mcdonnnj , update to fragment link. Addressed at 077cfbf

mcdonnnj commented on project_setup/skeletonize-existing-repository.md:
```diff
@@ -98,17 +101,17 @@ from merging in destructive upstream changes.
 - Update `setup.py` with non-example information
 - Arrange into appropriate folders, such as `src` and `test`
-- Update the `codeowners` to reflect subject matter expertise and
+- Update the `codeowners` file to reflect subject matter expertise and
```
```suggestion
- Update the `.github/CODEOWNERS` file to reflect subject matter expertise and
```

dv4harr10 replied:
Hi @mcdonnnj , text wording update. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

dv4harr10 committed:
Suggested text changes from code review

This commit updates Development guide
Readme and Skeletonize existing
repository according to requested
changes for wording, grammar,
replacing text and delete emoji.

dv4harr10 committed:
Update links per code review

This commit updates Development guide
links using a reference style for
gh-skeleton links in the Readme file
and directly links to the skeleton list
for skeletonize existing repository file.

dv4harr10 committed:
Update to Readme fragment links

This commit fixes some fragment links
that were not working.

dv4harr10 committed:
Add suggested link and grammar changes

Updates to add reference links,
fragment link, and text changes
to skeleton list file.

dv4harr10 committed:
Add text change to ReadMe

command promt dollar sign text change

dv4harr10 committed:
Add reference link to skeleton list file

Adding reference link to maintain
consistency of link references globally for
gh-skeleton extension tool.

dav3r commented on project_setup/README.md:
````diff
@@ -29,35 +28,34 @@ branch protection - [see our branch protection guide for details](branch-protect
-```
-
 ```console
+$ gh skeleton list
+
+
+
````
As long as we are updating this section, we should also update to show the current correct output:

```console
$ gh skeleton list
Available skeletons in cisagov:

skeleton-ansible-role    	A skeleton project for quickly getting a new cisagov Ansible role started.
skeleton-ansible-role-with-test-user	A skeleton project for quickly getting a new cisagov Ansible role started when that role requires an AWS test user.
skeleton-aws-lambda-python	A skeleton project for quickly getting a new Python-based AWS Lambda project started.
skeleton-docker          	A skeleton project for quickly getting a new cisagov Docker container started.
skeleton-generic         	A generic skeleton project for quickly getting a new cisagov project started.
skeleton-packer          	A skeleton project for quickly getting a new cisagov packer project started.
skeleton-python-library  	A skeleton project for quickly getting a new cisagov Python library started.
skeleton-tf-module       	A skeleton project for quickly getting a new cisagov Terraform module started.
```

dv4harr10 replied:
Hi @dav3r, update to output for gh skeleton list. Addressed at 3443b94

dav3r replied:
@dv4harr10 Why did you replace the whitespace between the skeleton repo names and their descriptions (in my sample output above) with newlines in https://github.com/cisagov/development-guide/commit/3443b94a4c2e3967ecd5407277d69abf1578accb?  Our formatting inside the `console` block should look the same as the output that users will see in their terminal.

dv4harr10 replied:
Hi @dav3r, the whitespace was giving a linter error. I can try to include it back.

dav3r replied:
Can you please show us exactly what error message you were getting?

dv4harr10 replied:
Hi @dav3r , it's resolved with 2be365c

dv4harr10 replied:
The error message was trim trailing whitespace, but it's now resolved.

dav3r replied:
Unfortunately, https://github.com/cisagov/development-guide/commit/2be365cadb2870773571224402ed0f7ae20c2320 still does not address my original concern:

> Why did you replace the whitespace between the skeleton repo names and their descriptions (in my sample output above) with newlines in https://github.com/cisagov/development-guide/commit/3443b94a4c2e3967ecd5407277d69abf1578accb?

dv4harr10 replied:
Hi @dav3r , please give me a call, I posted my number to the chat Thanks

dv4harr10 replied:
Hi @dav3r, update to output for gh skeleton list should be resolved at 8513d28

dav3r replied:
@dv4harr10 Looks good now with https://github.com/cisagov/development-guide/pull/97/commits/8513d287209b2f0c4ab81d68af244759eea753b9, thanks!

dv4harr10 committed:
Update output for gh skeleton list command

Referenced skeleton list required update.

dv4harr10 committed:
Update output for gh skeleton list command #2

add back whitespace to reflect what the user
should expect to see in terminal output.

dv4harr10 committed:
Update output for gh skeleton list command #3

Redo changes to formatting to reflect what the
user should expect to see in terminal output.
